src/SettingsMigrator/Migrator.cs

```diff
@@ -0,0 +1,276 @@
+            using (XmlTextReader reader = new XmlTextReader(oldRunsettingsPath))
+            {
+                reader.Namespaces = false;
+
+                var document = new XmlDocument();
+                document.Load(reader);
+                var root = document.DocumentElement;
+
+                var testsettingsNode = root.SelectSingleNode(@"/RunSettings/MSTest/SettingsFile");
+
+                if (testsettingsNode != null)
+                {
+                    testsettingsPath = testsettingsNode.InnerText;
+                }
+                if (!string.IsNullOrWhiteSpace(testsettingsPath))
+                {
+                    // Expand path relative to runsettings location.
+                    if (!Path.IsPathRooted(testsettingsPath))
+                    {
+                        testsettingsPath = Path.Combine(oldRunsettingsPath, testsettingsPath);
+                    }
+
+                    MigrateTestsettings(testsettingsPath, newRunsettingsPath, File.ReadAllText(oldRunsettingsPath));
+                }
+                else
+                {
+                    Console.WriteLine("Runsettings does not contain an embedded testsettings, not migrating.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Given a testsettings, converts it to runsettings
+        /// </summary>
+        /// <param name="testsettingsPath"></param>
+        /// <param name="newRunsettingsPath"></param>
+        /// <param name="oldRunSettingsContent"></param>
+        public void MigrateTestsettings(string testsettingsPath, string newRunsettingsPath, string oldRunSettingsContent = null)
+        {
+            using (XmlTextReader reader = new XmlTextReader(testsettingsPath))
+            {
+                reader.Namespaces = false;
+
+                var document = new XmlDocument();
+                document.Load(reader);
+                var root = document.DocumentElement;
+
+                // Select the interesting nodes from the xml.
+                var deploymentNode = root.SelectSingleNode(@"/TestSettings/Deployment");
+                var scriptnode = root.SelectSingleNode(@"/TestSettings/Scripts");
+                var websettingsNode = root.SelectSingleNode(@"/TestSettings/Execution/TestTypeSpecific/WebTestRunConfiguration");
+                var oldDatacollectorNodes = root.SelectNodes(@"/TestSettings/AgentRule/DataCollectors/DataCollector");
+                var timeoutNode = root.SelectSingleNode(@"/TestSettings/Execution/Timeouts");
+                var assemblyresolutionNode = root.SelectSingleNode(@"/TestSettings/Execution/TestTypeSpecific/UnitTestRunConfig");
+                var hostsNode = root.SelectSingleNode(@"/TestSettings/Execution/Hosts");
+                var executionNode = root.SelectSingleNode(@"/TestSettings/Execution");
+
+                string testTimeout = null;
+                if (timeoutNode != null && timeoutNode.Attributes[TestTimeoutAttributeName] != null)
```

Comment:
Add warning for deployment and script timeout

Reply:
Done